src/order/compactly_generated.lean

```diff
@@ -228,6 +228,25 @@ alias is_Sup_finite_compact_iff_is_sup_closed_compact ↔
       _ is_sup_closed_compact.is_Sup_finite_compact
 alias is_sup_closed_compact_iff_well_founded ↔ _ well_founded.is_sup_closed_compact
 
+lemma well_founded.finite_of_set_independent (h : well_founded ((>) : α → α → Prop))
```

Comment:
Situations where `>` is well-founded rather than `<` seem really rare to me. Does the dual statement make any sense (i.e. is there a dual notion for `set_independent`)?

Reply:
I think you might be thinking about this backwards.

A standard situation where `>` is well-founded is the lattice of submodules of Noetherian module. See [here](https://leanprover-community.github.io/mathlib_docs/ring_theory/noetherian.html#is_noetherian_iff_well_founded) for example.

You might also be interested in [this characterisation](https://leanprover-community.github.io/mathlib_docs/order/order_iso_nat.html#well_founded.monotone_chain_condition).

Reply:
@vihdzp, is `[is_well_order α (>)]` equivalent here or is it overly strong?

cc @urkud since you changed from one to the other in #14389

Reply:
If so, then 
```suggestion
lemma set_independent.finite [is_well_order α (>)]
```